### PR TITLE
[SYCL][XPTI] Add support for tracing Level Zero API calls

### PR DESCRIPTION
**Pull request by smaslov-intel**
Fork of https://github.com/intel/llvm/pull/5796

> Add support for tracing Level Zero API calls to Level Zero plugin. Also modify sycl-trace tool to be able to display both PI and L0 calls simultaneously.

### CODE REVIEW

**Review by pvchupin — DISMISSED** [review]
doc ok

File: sycl/plugins/level_zero/CMakeLists.txt
Comment (smaslov-intel) [inline sycl/plugins/level_zero/CMakeLists.txt:105]:
```suggestion
find_package(Threads REQUIRED)
```

**Comment by smaslov-intel** [comment]
@againull : please review

**Review by againull — APPROVED** [review]
Overall looks good to me.

File: sycl/plugins/level_zero/tracing.cpp
Comment (againull) [inline sycl/plugins/level_zero/tracing.cpp:142]: Should we call this function in piTearDown?

File: sycl/plugins/level_zero/CMakeLists.txt
Comment (againull) [inline sycl/plugins/level_zero/CMakeLists.txt:140]: I think we should put this under if (SYCL_ENABLE_XPTI_TRACING) too.

File: sycl/plugins/level_zero/CMakeLists.txt
Comment (againull) [inline sycl/plugins/level_zero/CMakeLists.txt:142]: I think we should put this under if (SYCL_ENABLE_XPTI_TRACING) too.

File: sycl/plugins/level_zero/CMakeLists.txt
Comment (againull) [inline sycl/plugins/level_zero/CMakeLists.txt:113]: I think we should include tracing.cpp into pi_level_zero.so only if (SYCL_ENABLE_XPTI_TRACING) is true.
Otherwise even release version of the library distributed to customers will contain unnecessary code which is used for debugging.

**Review by againull — CHANGES_REQUESTED** [review]

**Comment by againull** [comment]
Sorry, I've realized that I have a couple of concerns, so dimissed the approval.

File: sycl/plugins/level_zero/tracing.cpp
Comment (smaslov-intel) [reply sycl/plugins/level_zero/tracing.cpp:142]: Yes, I think we should. Added.

File: sycl/plugins/level_zero/CMakeLists.txt
Comment (smaslov-intel) [reply sycl/plugins/level_zero/CMakeLists.txt:113]: Actually I think we should do the opposite, i.e. remove the SYCL_ENABLE_XPTI_TRACING and always include the tracing capabilities. The "sycl-prof" and "sycl-trace" tools are meant to be available to the end-user, and should always produce the traces. FWIW, I already removed the SYCL_PI_LEVEL_ZERO_ENABLE_TRACING (made it to "1" always). Do you agree?

Comment (againull) [reply sycl/plugins/level_zero/CMakeLists.txt:113]: I see. Yes, I agree.

File: sycl/plugins/level_zero/tracing.cpp
Comment (againull) [reply sycl/plugins/level_zero/tracing.cpp:142]: Sergey, could you please push this update.

**Review by againull — APPROVED** [review]

**Comment by againull** [comment]
Looks good, but pls fix clang-format issue.

**Comment by pvchupin** [comment]
@smaslov-intel, @againull, it breaks Windows in post-commit: https://github.com/intel/llvm/runs/6164928984?check_suite_focus=true

**Comment by yanfeng3721** [comment]
@DoyleLi report Windows nightly build blocked by this PR, the issue was also detected in pre-commit test: http://icl-jenkins2.sc.intel.com:8080/blue/organizations/jenkins/SYCL_CI%2Fintel%2FWin%2FTest_Suite/detail/Test_Suite/1913/pipeline/
```
[2022-04-23T02:01:46.403Z] CMake Error at W:/jenkins-dir/workspace/SYCL_CI/intel/Win/Test_Suite/llvm.src/sycl/cmake/modules/AddSYCL.cmake:8 (add_library):

[2022-04-23T02:01:46.403Z]   Target "pi_level_zero" links to target "Threads::Threads" but the target

[2022-04-23T02:01:46.403Z]   was not found.  Perhaps a find_package() call is missing for an IMPORTED

[2022-04-23T02:01:46.403Z]   target, or an ALIAS target is missing?

[2022-04-23T02:01:46.403Z] Call Stack (most recent call first):

[2022-04-23T02:01:46.403Z]   W:/jenkins-dir/workspace/SYCL_CI/intel/Win/Test_Suite/llvm.src/sycl/cmake/modules/AddSYCL.cmake:41 (add_sycl_library)

[2022-04-23T02:01:46.403Z]   W:/jenkins-dir/workspace/SYCL_CI/intel/Win/Test_Suite/llvm.src/sycl/plugins/level_zero/CMakeLists.txt:106 (add_sycl_plugin)
```

**Comment by smaslov-intel** [comment]
> @smaslov-intel, @againull, it breaks Windows in post-commit: https://github.com/intel/llvm/runs/6164928984?check_suite_focus=true

trying to fix in https://github.com/intel/llvm/pull/6059

**Pull request merged**
